stdlib/public/core/Result.swift

```diff
@@ -0,0 +1,202 @@
+  ///
+  /// Use the `map` method with a closure that returns a non-`Result` value.
+  ///
+  /// - Parameter transform: A closure that takes the successful value of the
+  ///   instance.
+  /// - Returns: A new `Result` instance with the result of the transform, if
+  ///   it was applied.
+  public func map<NewValue>(
+    _ transform: (Value) -> NewValue
+  ) -> Result<NewValue, Error> {
+    switch self {
+    case let .success(value): return .success(transform(value))
+    case let .failure(error): return .failure(error)
+    }
+  }
+  
+  /// Evaluates the given transform closure when this `Result` instance is
+  /// `.failure`, passing the error as a parameter.
+  ///
+  /// Use the `mapError` method with a closure that returns a non-`Result`
+  /// value.
+  ///
+  /// - Parameter transform: A closure that takes the failure value of the
+  ///   instance.
+  /// - Returns: A new `Result` instance with the result of the transform, if
+  ///   it was applied.
+  public func mapError<NewError>(
+    _ transform: (Error) -> NewError
+  ) -> Result<Value, NewError> {
+    switch self {
+    case let .success(value): return .success(value)
+    case let .failure(error): return .failure(transform(error))
+    }
+  }
+  
+  /// Evaluates the given transform closure when this `Result` instance is
+  /// `.success`, passing the value as a parameter and flattening the result.
+  ///
+  /// - Parameter transform: A closure that takes the successful value of the
+  ///   instance.
+  /// - Returns: A new `Result` instance, either from the transform or from
+  ///   the previous error value.
+  public func flatMap<NewValue>(
+    _ transform: (Value) -> Result<NewValue, Error>
+  ) -> Result<NewValue, Error> {
+    switch self {
+    case let .success(value): return transform(value)
+    case let .failure(error): return .failure(error)
+    }
+  }
+  
+  /// Evaluates the given transform closure when this `Result` instance is
+  /// `.failure`, passing the error as a parameter and flattening the result.
+  ///
+  /// - Parameter transform: A closure that takes the error value of the
+  ///   instance.
+  /// - Returns: A new `Result` instance, either from the transform or from
+  ///   the previous success value.
+  public func flatMapError<NewError>(
+    _ transform: (Error) -> Result<Value, NewError>
+  ) -> Result<Value, NewError> {
+    switch self {
+    case let .success(value): return .success(value)
+    case let .failure(error): return transform(error)
+    }
+  }
+  
+  /// Evaluates the given transform closures to create a single output value.
+  ///
+  /// - Parameters:
+  ///   - onSuccess: A closure that transforms the success value.
+  ///   - onFailure: A closure that transforms the error value.
+  /// - Returns: A single `Output` value.
+  public func fold<Output>(
+    onSuccess: (Value) -> Output,
+    onFailure: (Error) -> Output
+  ) -> Output {
+    switch self {
+    case let .success(value): return onSuccess(value)
```

Comment:
Wrap to next line here and in the next line.
```suggestion
    case let .success(value):
      return onSuccess(value)
```